File: vripper-server/src/main/java/tn/mnlr/vripper/services/ThreadPoolService.java
package tn.mnlr.vripper.services;

import lombok.Getter;
import org.springframework.stereotype.Service;

import java.util.concurrent.ExecutorService;
import java.util.concurrent.Executors;
import java.util.concurrent.TimeUnit;

@Service
public class ThreadPoolService {
  @Getter
  private final ExecutorService generalExecutor =
      Executors.newFixedThreadPool(Runtime.getRuntime().availableProcessors());

  public void destroy() throws Exception {
    generalExecutor.shutdown();
    generalExecutor.awaitTermination(5, TimeUnit.SECONDS);
  }
}

File: vripper-core/src/main/kotlin/me/mnlr/vripper/services/ThreadPoolService.kt
package me.mnlr.vripper.services

import org.springframework.stereotype.Service
import java.util.concurrent.ExecutorService
import java.util.concurrent.Executors
import java.util.concurrent.TimeUnit

@Service
class ThreadPoolService {
    val generalExecutor: ExecutorService = Executors.newFixedThreadPool(4)
    fun destroy() {
        generalExecutor.shutdown()
        if (!generalExecutor.awaitTermination(5, TimeUnit.SECONDS)) {
            generalExecutor.shutdownNow()
        }
    }
}